airbyte-ci/connectors_ci/metadata_service/orchestrator/orchestrator/resources/gcp_resources.py

```diff
@@ -0,0 +1,86 @@
+import json
+
+from google.cloud import storage
+from google.oauth2 import service_account
+
+from dagster import StringSource, InitResourceContext, resource
+from dagster_gcp.gcs.file_manager import GCSFileManager
+
+
+@resource(config_schema={"gcp_gsm_cred_string": StringSource})
+def gcp_gcs_client(resource_context: InitResourceContext) -> storage.Client:
```

Comment:
What's the conceptual difference between *assets* and *resources*?

Reply:
Its a hard one to explain fully. 

To me it seems that:
1. asset means the function becomes a node in the dag and its inputs and outputs recorded to disk
2. A resource is any class or method you need to interact with the outside world in service of creating assets